unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Tests/BodyshapeLoaderShould.cs

```diff
@@ -0,0 +1,335 @@
+
+            //Act
+            Color hairColor = Color.red;
+            Color skinColor = Color.blue;
+            Color eyesColor = Color.yellow;
+            await bodyshapeLoader.Load(container, new AvatarSettings
+            {
+                bodyshapeId = WearableLiterals.BodyShapes.MALE,
+                hairColor = Color.red,
+                skinColor = Color.blue,
+                eyesColor = Color.yellow
+            });
+
+            //Assert
+            Assert.AreEqual(IWearableLoader.Status.Succeeded, bodyshapeLoader.status);
+            Assert.AreEqual(Color.gray, bodyParts.head.material.color);
+            Assert.AreEqual(hairColor, bodyParts.ubody.material.color);
+            Assert.AreEqual(skinColor, bodyParts.lbody.material.color);
+            Assert.AreEqual(Color.gray, bodyParts.feet.material.color);
+            Assert.AreEqual(eyesColor, bodyParts.eyes.material.GetColor(AvatarSystemUtils._EyeTint));
+            Assert.AreEqual(Texture2D.whiteTexture, bodyParts.eyes.material.GetTexture(AvatarSystemUtils._EyesTexture));
+            Assert.AreEqual(hairColor, bodyParts.eyebrows.material.GetColor(AvatarSystemUtils._BaseColor));
+            Assert.AreEqual(Texture2D.whiteTexture, bodyParts.eyebrows.material.GetTexture(AvatarSystemUtils._BaseMap));
+            Assert.AreEqual(skinColor, bodyParts.mouth.material.GetColor(AvatarSystemUtils._BaseColor));
+            Assert.AreEqual(Texture2D.whiteTexture, bodyParts.mouth.material.GetTexture(AvatarSystemUtils._BaseMap));
+        });
+
+        [UnityTest]
+        public IEnumerator ThrowsIfCanceledTokenIsPassed() => UniTask.ToCoroutine(async () =>
+        {
+            //Arrange
+            var cts = new CancellationTokenSource();
+            cts.Cancel();
+
+            // Act & Assert
+            await TestUtils.ThrowsAsync<OperationCanceledException>(bodyshapeLoader.Load(container, new AvatarSettings(), cts.Token));
+        });
+
+        [UnityTest]
+        public IEnumerator ThrowsIfFailingBodyshape_FromThrow() => UniTask.ToCoroutine(async () =>
+        {
+            //Arrange
+            bodyshapeLoader.bodyshapeRetriever.Configure()
+                           .Retrieve(Arg.Any<GameObject>(), Arg.Any<ContentProvider>(), Arg.Any<string>(), Arg.Any<string>(), Arg.Any<CancellationToken>())
+                           .Returns(x => throw new Exception("ThrowingOnPurpose"));
+
+            //Assert
+            await TestUtils.ThrowsAsync<Exception>(bodyshapeLoader.Load(container, new AvatarSettings()));
+            bodyshapeLoader.bodyshapeRetriever.Received().Dispose();
+            bodyshapeLoader.eyesRetriever.Received().Dispose();
+            bodyshapeLoader.eyebrowsRetriever.Received().Dispose();
+            bodyshapeLoader.mouthRetriever.Received().Dispose();
+        });
+
+        [UnityTest]
+        public IEnumerator ThrowsIfFailingBodyshape_FromNull() => UniTask.ToCoroutine(async () =>
```

Comment:
I would keep the `Throw` because it's the main point of this test, to assert that the process throws an exception. 

What about: `ThrowWhenBodyshapeRetrieverFails`?

Reply:
sounds good